packages/wonder-blocks-tooltip/components/tooltip-anchor.js

```diff
@@ -0,0 +1,168 @@
+            );
+        }
+
+        this._anchorNode = anchorNode;
+        this._updateFocusivity();
+        if (anchorNode) {
+            // TODO(somewhatabstract): Detect ESC key to deactivate
+            // TODO(somewhatabstract): Work out how to allow pointer to go over
+            // the tooltip content to keep it active.
+
+            anchorNode.addEventListener("focusin", () => this._handleFocusIn());
+            anchorNode.addEventListener("focusout", () =>
+                this._handleFocusOut(),
+            );
+            anchorNode.addEventListener("mouseenter", () =>
+                this._handleMouseEnter(),
+            );
+            anchorNode.addEventListener("mouseleave", () =>
+                this._handleMouseLeave(),
+            );
+
+            this.props.anchorRef(this._anchorNode);
+        }
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (
+            prevProps.forceAnchorFocusivity !==
+                this.props.forceAnchorFocusivity ||
+            prevProps.children !== this.props.children
+        ) {
+            this._updateFocusivity();
+        }
+    }
+
+    componentWillUnmount() {
+        const anchorNode = this._anchorNode;
+        if (anchorNode) {
+            anchorNode.removeEventListener("focusin", () =>
+                this._handleFocusIn(),
+            );
+            anchorNode.removeEventListener("focusout", () =>
+                this._handleFocusOut(),
+            );
+            anchorNode.removeEventListener("mouseenter", () =>
+                this._handleMouseEnter(),
+            );
+            anchorNode.removeEventListener("mouseleave", () =>
+                this._handleMouseLeave(),
+            );
+        }
+    }
+
+    _updateFocusivity() {
+        const anchorNode = this._anchorNode;
+        if (!anchorNode) {
+            return;
+        }
+        const {forceAnchorFocusivity} = this.props;
+        const currentTabIndex = anchorNode.getAttribute("tabindex");
+
+        if (forceAnchorFocusivity && !currentTabIndex) {
+            // Ensure that the anchor point is keyboard focusable so that
+            // we can show the tooltip for visually impaired users that don't
+            // use pointer devices nor assistive technology like screen readers.
+            anchorNode.setAttribute("tabindex", "0");
+            this._weSetFocusivity = true;
+        } else if (!forceAnchorFocusivity && currentTabIndex) {
+            // We may not be forcing it, but we also want to ensure that if we
+            // did before, we remove it.
+            if (this._weSetFocusivity) {
+                anchorNode.removeAttribute("tabindex");
+                this._weSetFocusivity = false;
+            }
+        }
+    }
+
+    _updateActiveState(hovered: boolean, focused: boolean) {
+        // Take a snapshot of the old and new state.
+        const oldState = this.state.active;
+        const newState = !!(hovered || focused);
```

Comment:
The `!!` isn't necessary since `hovered` and `focused` are both booleans.

Reply:
Yeah, they didn't used to be :grin: Hangover of old code.